### PR TITLE
refactor: replace glog in global_configuration.go

### PR DESCRIPTION
**Pull request by AlexFenlon**
### Proposed changes

- remove `google/glog` from main.go

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [ ] I have rebased my branch onto main
- [ ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by pdabelf5 — APPROVED** [review]

**Review by jjngx — APPROVED** [review]

File: internal/k8s/global_configuration.go [hunk 1]
```diff
@@ -16,30 +16,30 @@ func createGlobalConfigurationHandlers(lbc *LoadBalancerController) cache.Resour
 	return cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			gc := obj.(*conf_v1.GlobalConfiguration)
-			glog.V(3).Infof("Adding GlobalConfiguration: %v", gc.Name)
+			nl.Debugf(lbc.logger, "Adding GlobalConfiguration: %v", gc.Name)
```
Comment (j1m-ryan) [inline internal/k8s/global_configuration.go:19]: Why is this Debugf Instead of Infof?

File: internal/k8s/global_configuration.go [hunk 2]
```diff
@@ -74,11 +74,11 @@ func (lbc *LoadBalancerController) syncGlobalConfiguration(task task) {
 	var validationErr error
 
 	if !gcExists {
-		glog.V(2).Infof("Deleting GlobalConfiguration: %v\n", key)
+		nl.Debugf(lbc.logger, "Deleting GlobalConfiguration: %v\n", key)
```
Comment (j1m-ryan) [inline internal/k8s/global_configuration.go:77]: Should this be InfoF then? as its v2 previously

Comment (AlexFenlon) [reply internal/k8s/global_configuration.go:77]: according to the levels package 
```
	// LevelDebug - Debug Level Logging same as glog.V(2)
	LevelDebug = slog.LevelDebug
```

Comment (j1m-ryan) [reply internal/k8s/global_configuration.go:77]: So what does this mean for the V(3) ones?
Earlier this pr this
`glog.V(3).Infof("Adding GlobalConfiguration: %v", gc.Name)` 
changed to 
`nl.Debugf(lbc.logger, "Adding GlobalConfiguration: %v", gc.Name)`

And here a v2 Info
`glog.V(2).Infof("Deleting GlobalConfiguration: %v\n", key)`
is changing also to Debugf 
`nl.Debugf(lbc.logger, "Deleting GlobalConfiguration: %v\n", key)`

Is this intentional?

Comment (pdabelf5) [reply internal/k8s/global_configuration.go:77]: yep, there isn't a level between Debug and Info in logging libraries.  The alternative would be to have `glog.V(3)` as TRACE, and `glog.V(2)` as DEBUG

**Pull request merged**